medicines/api/src/document_type.rs

```diff
@@ -0,0 +1,78 @@
+use std::fmt::{Display, Formatter};
+use std::str::FromStr;
+
+#[derive(Debug, Copy, Clone, Eq, PartialEq, Ord, PartialOrd, juniper::GraphQLEnum)]
+pub enum DocumentType {
+    Spc,
+    Pil,
+    Par,
+}
+
+impl DocumentType {
+    pub fn to_search_str(&self) -> &str {
+        match self {
+            DocumentType::Spc => "Spc",
+            DocumentType::Pil => "Pil",
+            DocumentType::Par => "Par",
+        }
+    }
+}
+
+impl FromStr for DocumentType {
+    type Err = DocTypeParseError;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        match s.to_ascii_uppercase().as_str() {
+            "SPC" => Ok(Self::Spc),
+            "PIL" => Ok(Self::Pil),
+            "PAR" => Ok(Self::Par),
+            _ => Err(DocTypeParseError {
+                source: s.to_string(),
+            }),
+        }
+    }
+}
+
+impl Display for DocumentType {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
```

Comment:
This all seems familiar – I think we have similar in `doc-index-updater`. Given that it's all about reading from `IndexResult`, and we're doing this in every crate where we read `IndexResult`, could we move this `DocumentType` into `search-client`?